### PR TITLE
[Trivial] Use same multi-source-syntax for test files

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ Compiler Features:
  * SMTChecker: Support ``type(T).min`` and ``type(T).max``.
  * Yul Optimizer: Prune unused parameters in functions.
  * Yul Optimizer: Try to simplify function names.
+ * Tests: Use same syntax for multi-line ASTJSON tests as in other places
```
Comment (chriseth) [inline Changelog.md]: I don't think users care about this feature :)

File: test/libsolidity/ASTJSONTest.cpp [hunk 2]
```diff
@@ -90,7 +90,9 @@ ASTJSONTest::ASTJSONTest(string const& _filename)
 			if (!sourceName.empty())
 				m_sources.emplace_back(sourceName, source);
 
-			sourceName = line.substr(sourceDelimiter.size(), string::npos);
+			sourceName = line.substr(
+				sourceDelimiter.size(),
+				line.size() - string(" ====").size() - sourceDelimiter.size());
```
Comment (cameel) [inline test/libsolidity/ASTJSONTest.cpp]:
```suggestion
				line.size() - " ===="s.size() - sourceDelimiter.size());
```

File: test/libsolidity/ASTJSONTest.cpp [hunk 3]
```diff
@@ -237,8 +239,6 @@ void ASTJSONTest::printSource(ostream& _stream, string const& _linePrefix, bool
 {
 	for (auto const& source: m_sources)
 	{
-		if (m_sources.size() > 1 || source.first != "a")
-			_stream << _linePrefix << "// ---- SOURCE: " << source.first << endl << endl;
```
Comment (cameel) [inline test/libsolidity/ASTJSONTest.cpp]: With this removed, `isoltest` prints the source without delimiters, so it looks like a single file:

``` solidity
    /**This contract is empty*/ contract C {}

    /**This contract is empty
     and has a line-breaking comment.*/
    contract C {}

    contract C {
      /** Some comment on state var.*/ uint public state;
      /** Some comment on Evt.*/ event Evt();
      /** Some comment on mod.*/ modifier mod() { _; }
      /** XXX*/ function fn() public {}
    }
```
Note: I replaced the last comment with `XXX` to have the test fail.

File: test/libsolidity/ASTJSONTest.cpp [hunk 4]
```diff
@@ -90,7 +90,9 @@ ASTJSONTest::ASTJSONTest(string const& _filename)
 			if (!sourceName.empty())
 				m_sources.emplace_back(sourceName, source);
 
-			sourceName = line.substr(sourceDelimiter.size(), string::npos);
+			sourceName = line.substr(
+				sourceDelimiter.size(),
+				line.size() - " ===="s.size() - sourceDelimiter.size());
```
Comment (chriseth) [inline test/libsolidity/ASTJSONTest.cpp]:
```suggestion
				line.size() - " ===="s.size() - sourceDelimiter.size()
			);
```

File: test/libsolidity/ASTJSONTest.cpp (hunk 3 above)
Comment (Marenz) [reply test/libsolidity/ASTJSONTest.cpp]: Good point, I'll add it back and adjust it

**Review by cameel — APPROVED** [review]

**Pull request merged**